libshaderc/include/shaderc/shaderc.hpp

```diff
@@ -55,25 +72,30 @@ class CompilationResult {
     return shaderc_result_get_compilation_status(compilation_result_);
   }
 
-  // Returns a pointer to the start of the compiled SPIR-V.
-  // It is guaranteed that static_cast<uint32_t> is valid to call on this
-  // pointer.
-  // This pointer remains valid only until the CompilationResult is destroyed.
-  const char* GetData() const {
-    if (!compilation_result_) {
-      return "";
-    }
-    return shaderc_result_get_bytes(compilation_result_);
+  // Returns an iterator pointing to the start of the compilation
```

Comment:
Can we specify that this is a random_access_iterator.

We should also mention that the iterator returned iterates through contiguous memory, which is an important distinction if we plan on being able to put this in a string_piece.